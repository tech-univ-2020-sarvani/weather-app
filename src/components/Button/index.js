import React from 'react';
import PropTypes from 'prop-types';
// import * as styles from './index.css';
import './index.css';
import { Link, BrowserRouter as Router } from 'react-router-dom';

const Button = (props) => {
  const { children, testId, path } = props;
  return (
    <div className="buttonContainer">
      <Router>
        <Link to={path}>
          <button className="button" type="button" data-testid={testId}>{children}</button>
        </Link>
      </Router>
    </div>
  );
};

Button.propTypes = {
  children: PropTypes.string.isRequired,
  testId: PropTypes.string,
  path: PropTypes.string,
};

export default Button;
