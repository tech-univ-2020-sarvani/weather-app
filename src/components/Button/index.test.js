import React from 'react';
import { render } from '@testing-library/react';
import Button from './index';

describe('The button component', () => {
  it('shouls render correctly', () => {
    const { asFragment } = render(<Button testId="test-btn" path="/">Test-text</Button>);
    expect(asFragment()).toMatchSnapshot();
  });
});
