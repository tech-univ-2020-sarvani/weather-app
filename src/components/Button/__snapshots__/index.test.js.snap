// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The button component shouls render correctly 1`] = `
<DocumentFragment>
  <div
    class="button"
  >
    <button
      data-testid="test-btn"
      type="button"
    >
      Test-text
    </button>
  </div>
</DocumentFragment>
`;
